File: backend/api/src/main/java/com/github/ai/leetcodequiz/api/response/GetUnansweredQuestionsResponse.java
package com.github.ai.leetcodequiz.api.response;

import com.github.ai.leetcodequiz.api.QuestionItemDto;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public record GetUnansweredQuestionsResponse(
    @NotNull List<QuestionItemDto> questions
) {
}

File: backend/api/src/main/scala/com/github/ai/leetcodequiz/api/response/GetUnansweredQuestionsResponse.scala
package com.github.ai.leetcodequiz.api.response

import com.github.ai.leetcodequiz.api.QuestionItemDto
import zio.json.{JsonDecoder, JsonEncoder}

case class GetUnansweredQuestionsResponse(
  questions: List[QuestionItemDto]
) derives JsonEncoder, JsonDecoder
